tensorflow/compiler/mlir/xla/transforms/buffer_assignment.cc

```diff
@@ -463,8 +463,9 @@ void FunctionAndBlockSignatureConverter::addDynamicallyLegalFuncOp(
     ConversionTarget& target) {
   target.addDynamicallyLegalOp<FuncOp>([&](FuncOp op) {
     auto inputs = op.getType().getInputs();
-    return std::all_of(inputs.begin(), inputs.end(),
-                       [](Type input) { return input.isa<MemRefType>(); });
+    return std::all_of(inputs.begin(), inputs.end(), [](Type input) {
+      return input.isa<MemRefType>() || !input.isa<ShapedType>();
+    });
```

Comment:
You should update the command above this block of code, also document why instead of just what the code is doing as possible.